### PR TITLE
[SPARK-29906][SQL][FOLLOWUP] Update the final plan in UI for AQE

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

a followup of https://github.com/apache/spark/pull/26576, which mistakenly removes the UI update of the final plan.

### Why are the changes needed?

fix mistake.

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @maryannxue @JkSelf

**Comment by JkSelf** [comment]
LGTM. Thanks

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.

**Comment by manuzhang** [comment]
@cloud-fan Is there any UT to catch such mistake that it won't happen again ?

**Comment by cloud-fan** [comment]
unfortunately we don't have a good test framework for UI. It would be great if we have one.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala [hunk 1]
```diff
@@ -205,6 +205,7 @@ case class AdaptiveSparkPlanExec(
 
       // Run the final plan when there's no more unfinished stages.
       currentPhysicalPlan = applyPhysicalRules(result.newPlan, queryStageOptimizerRules)
+      executionId.foreach(onUpdatePlan)
```
Comment (manuzhang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: Why is the plan sent out before `isFinalPlan = true` ? Isn't it the final plan ?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: It doesn't matter. `onUpdatePlan` only look at `currentPhysicalPlan`, so we must call `onUpdatePlan` after setting `currentPhysicalPlan`.

Comment (manuzhang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: Does it matter for what's displayed on UI or anyone interested in sql update event ?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: what do you mean? Even if you move `isFinalPlan = true` before this line, nothing gets changed.

Comment (manuzhang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: The root nodes of `Physical Plan` in `physicalPlanDescription`s  are different: `AdaptiveSparkPlan(isFinalPlan=true)` vs `AdaptiveSparkPlan(isFinalPlan=false)`. So are root nodes of `sparkPlan`

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: ah good catch! somehow I misread the code and thought it doesn't matter.

For UI it doesn't matter as we will strip the `AdaptiveSparkPlanExec`, but if there is a spark listener catching this event, it matters.

Can you send a PR to fix it? Thanks!

Comment (manuzhang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: Sure. Thanks for your reply.

Comment (manuzhang) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala]: I've sent https://github.com/apache/spark/pull/26983. Please review.

**Pull request closed without merging**